src/dos/drive_virtual.cpp

```diff
@@ -198,18 +209,18 @@ void VFILE_Remove(const char *name, const char *dir = "")
 		if (onpos == 0)
 			return;
 	}
-	auto vfile = first_file;
-	while (vfile) {
-		if (onpos == vfile->onpos && vfile->name == name) {
-			if (vfile.get() == first_file.get()) {
-				first_file = vfile->next;
-			}
-			// Finally release the vfile itself
-			vfile.reset();
-
-			return;
+	auto vfile = find_vfile_by_predicate(first_file,
+	                                     [onpos, name](vfile_block_t vfile) {
+		return onpos == vfile->onpos && vfile->name == name;
+	});
+	if (vfile) {
```

Comment:
```c++
find_vfile_by_predicate(first_file,
	                                     [onpos, name](vfile_block_t vfile) {
		return onpos == vfile->onpos && vfile->name == name;
	});
```

I'm still finding the inline lamda definition (which itself is passed into the predicate routine) to be non-trivial to read fluidly.

Similar to:
```c++
if (vfile_name_and_pos_exists(name, onpos)) {
	return;
}
```

It would be nice to have a wrapper here:
```c++
auto vfile = find_vfile_by_name_and_pos(name, onpos);
```

This could even simplify `vfile_name_and_pos_exists`:
```c++
bool vfile_name_and_pos_exists(...) {
	// use the bool operator to test the result
	return find_vfile_by_name_and_pos(...);
}
```

Reply:
problem here: the lambda is using case-sensitive file name comparison - instead of iequals usage in the current helper - i don't know if that makes a difference

Reply:
Good catch, @LowLevelMahn ; these little case differences weren't obviously in the original code due to all the gnashing about manually cranking through the linked list and doing these rote comparisons. 

I think this revealed an oversight in the original code. 

If we have VFILE_Register() doing an insensitive comparison for dupes, prior to creation, then I think it's fair to similarly run an insensitive comparison for Re-registration. 

DOS was always case-insensitive, so I think reusing the insensitive comparison is the correct approach.